crates/ruma-events/src/room/message.rs

```diff
@@ -669,6 +689,24 @@ impl FormattedBody {
     pub fn html(body: impl Into<String>) -> Self {
         Self { format: MessageFormat::Html, body: body.into() }
     }
+
+    /// Creates a new HTML-formatted message body by parsing the markdown in `body`.
+    ///
+    /// Returns None if no markdown formatting was found.
+    #[cfg(feature = "markdown")]
+    #[cfg_attr(docsrs, doc(cfg(feature = "markdown")))]
+    pub fn markdown(body: impl Into<String>) -> Option<Self> {
+        let body = body.into();
+        let mut html_body = String::new();
+
+        pulldown_cmark::html::push_html(&mut html_body, pulldown_cmark::Parser::new(&body));
+
+        if html_body == format!("<p>{}</p>\n", body) {
```

Comment:
I see an allocation that's not needed! But `html_body.starts_with("<p>") && html_body[3..].starts_with(body) && html_body[3 + body.len()..] == "</p>\n"` is a mouthful, so probably better to keep this as-is 😅